src/v/storage/mvlog/versioned_log.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "storage/mvlog/versioned_log.h"
+
+#include "container/fragmented_vector.h"
+#include "model/offset_interval.h"
+#include "storage/mvlog/file.h"
+#include "storage/mvlog/logger.h"
+#include "storage/mvlog/readable_segment.h"
+#include "storage/mvlog/segment_appender.h"
+
+#include <seastar/core/circular_buffer.hh>
+#include <seastar/core/io_priority_class.hh>
+#include <seastar/core/lowres_clock.hh>
+
+#include <chrono>
+
+using namespace std::literals::chrono_literals;
+
+namespace storage::experimental::mvlog {
+
+active_segment::active_segment(
+  std::unique_ptr<file> f, model::offset o, segment_id id, size_t target_size)
+  : segment_file(std::move(f))
+  , appender(std::make_unique<segment_appender>(segment_file.get()))
+  , readable_seg(std::make_unique<readable_segment>(segment_file.get()))
+  , construct_time(ss::lowres_clock::now())
+  , target_max_size(target_size)
+  , id(id)
+  , base_offset(o)
+  , next_offset(o) {}
```

Comment:
is it empty when these values are equal?

Reply:
That's right